cases/deepkit.ts

```diff
@@ -0,0 +1,46 @@
+import { validate, deserialize, Negative } from '@deepkit/type';
+import { addCase } from '../benchmarks';
+
+interface DataType {
+  number: number;
+  negNumber: number & Negative;
+  maxNumber: number;
+  string: string;
+  longString: string;
+  boolean: boolean;
+  deeplyNested: {
+    foo: string;
+    num: number;
+    bool: boolean;
+  };
+}
+
+interface DataTypeSafe extends DataType {
+  [key: string]: unknown;
+  deeplyNested: {
+    foo: string;
+    num: number;
+    bool: boolean;
+    [key: string]: unknown;
+  };
+}
```

Comment:
@BasixKOR The spirit of this repo is to define types in code, and then infer the types from it AND validate.

If you have to define the interfaces by hand, then it is not suitable for this repository.

Reply:
IIRC deepkit uses this interface to generate the runtime types for validation. This is in fact a type definition to be used for validation.

Reply:
@BasixKOR that is exactly how it works and advanced runtime reflection is one big selling point of deepkit.

@moltar IMHO it follows the spirit of "you only have to define your types once" - interface and validator are always in sync.

Reply:
@BasixKOR what I don't get tough is `DataTypeSafe` interface - that does not look like it matches the test type at all - should you always just use the `DataType` or maybe `DataType` with `[key: string]: unknown;` to "ignore" additional keys?

Reply:
The latter one doesn't really corresponds to the test type too, so I think it's better to test against `strict` cases for now.

Reply:
Ah, great, didn't realize that. Sorry 😅